addons/l10n_fr_invoice_addr/models/account_move.py

```diff
@@ -0,0 +1,14 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    l10n_fr_is_company_french = fields.Boolean(compute='_compute_l10n_fr_is_company_french')
+
+    @api.depends('company_id.country_code')
+    def _compute_l10n_fr_is_company_french(self):
+        for record in self:
+            record.l10n_fr_is_company_french = (record.company_id.country_code in record.company_id._get_unalterable_country()) if record.company_id else False
```

Comment:
```suggestion
            record.l10n_fr_is_company_french = record.company_id.country_code == 'FR'
```
* I don't see what _get_unalterable_country has to do with the feature/requirement
* company_id is required on account.move
* browse_null record being browsed returns False anyway

Reply:
>- I don't see what _get_unalterable_country has to do with the feature/requirement

It returns the country codes of not only mainland France but also of the DOM-TOMs which also have their own country codes while being part of France. At least, that's my understanding after reading this piece of code:
https://github.com/odoo/odoo/blob/a3d0cec412edd2f0605c8ca5ed11f35ea1afe839/addons/l10n_fr/models/res_company.py#L18-L21

>- company_id is required on account.move
>- browse_null record being browsed returns False anyway

:+1:

Reply:
ok it's far from being trivial because of the function name that is totally misleading and now it's used for another purpose.

I'd suggest a small refactor to change function name (or give it an alias)